lib/base-compiler.js

```diff
@@ -367,15 +367,14 @@ class BaseCompiler {
 
     postProcessAsm(result) {
         if (!result.okToCache) return result;
-        const demangler = this.compiler.demangler;
-        if (!demangler) return result;
-        return this.exec(demangler, [], {input: _.pluck(result.asm, 'text').join("\n")})
-            .then((demangleResult) => {
-                const lines = utils.splitLines(demangleResult.stdout);
-                for (let i = 0; i < result.asm.length; ++i)
-                    result.asm[i].text = lines[i];
-                return result;
-            });
+        const demanglerExe = this.compiler.demangler;
+        if (!demanglerExe) return result;
+
+        const demanglerClassFile = this.compilerProps("demanglerClassFile");
+        const demanglerClass = require(demanglerClassFile).Demangler;
+
+        const demangler = new demanglerClass(demanglerExe, null, this);
```

Comment:
Could this be avoided to be done in every compilation request, by creating it at creation time?